hnn_core/parallel_backends.py

```diff
@@ -50,12 +50,17 @@ def _gather_trial_data(sim_data, net, n_trials, postproc):
     for idx in range(n_trials):
         dpls.append(sim_data[idx][0])
         spikedata = sim_data[idx][1]
-        net.cell_response._spike_times.append(spikedata[0])
-        net.cell_response._spike_gids.append(spikedata[1])
+        spike_times = spikedata[0]
+        spike_gids = spikedata[1]
         net.gid_ranges = spikedata[2]  # only have one gid_ranges
-        net.cell_response.update_types(net.gid_ranges)
-        net.cell_response._vsoma.append(spikedata[3])
-        net.cell_response._isoma.append(spikedata[4])
+        for cell_type, gid_range in net.gid_ranges.items():
+            if cell_type in net.cellname_list:
```

Comment:
why is this `if` clause needed?

Reply:
Sorry missed this comment, it's needed because cell response isn't supposed to contain gids associated with sources. I mean it can, but it's redundant information.